geode-core/src/integrationTest/java/org/apache/geode/internal/net/SSLSocketIntegrationTest.java

```diff
@@ -174,6 +189,129 @@ public void securedSocketTransmissionShouldWork() throws Exception {
     assertThat(this.messageFromClient.get()).isEqualTo(MESSAGE);
   }
 
+  @Test
+  public void testSecuredSocketTransmissionShouldWorkUsingNIO() throws Exception {
+    ServerSocketChannel serverChannel = ServerSocketChannel.open();
+    serverSocket = serverChannel.socket();
+
+    InetSocketAddress addr = new InetSocketAddress(localHost, 0);
+    serverSocket.bind(addr, 10);
+    int serverPort = this.serverSocket.getLocalPort();
+
+    SocketCreator clusterSocketCreator =
+        SocketCreatorFactory.getSocketCreatorForComponent(SecurableCommunicationChannel.CLUSTER);
+    this.serverThread = startServerNIO(serverSocket, 15000);
+
+    await().until(() -> serverThread.isAlive());
```

Comment:
There is still a tiny race window here in which the serverThread is alive but hasn't created the server socket yet. You might want to change this to actually attempt the connect within the await.